.github/workflows/lint.yml

```diff
@@ -99,12 +99,10 @@ jobs:
         with:
           script: |
             return [...new Set(${{ steps.filter.outputs.subworkflows_files }}
+              .filter(x => x.endsWith('main.nf'))
               .map(path => path
                 .replace('subworkflows/nf-core/', '')
```

Comment:
This is to fix the issue described in https://github.com/nf-core/modules/pull/7233#issuecomment-2771632041 where "Dockerfile" and the parent folder were considered a separate module. 

My logic for the new version is that every module must have a `main.nf`, therefore let's only consider those folders.